### PR TITLE
LPS-146731 & LPS-147053 Export PublishButton to other file and create tests for that

### PR DESCRIPTION
**Pull request by veroglez**

### CODE REVIEW

**Comment by veroglez** [comment]
ci:test:sf

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/contexts/StyleErrorsContext.js [hunk 1]
```diff
@@ -12,26 +12,76 @@
...
 	);
 }
 
-export function useSetHasStyleErrors() {
-	return useContext(StyleErrorsDispatchContext);
+export function useDeleteStyleError() {
+	const {setState, state} = useContext(StyleErrorsStateContext);
+
+	return useCallback(
+		(fieldName, itemId = 'defaultId') => {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/contexts/StyleErrorsContext.js:33]: Maybe it is safer to extract it to a `DEFAULT_ID` constant to prevent typos in the future

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/ColorPicker/ColorPicker.js [hunk 2]
```diff
@@ -145,9 +174,16 @@ export function ColorPicker({
 			});
 
 			if (nextValue.error) {
-				setHasStyleErrors(true);
 				setError(nextValue.error);
 				setCustomColors(['FFFFFF']);
+				setStyleError(
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/ColorPicker/ColorPicker.js:154]: Sometimes I hate prettier :(

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/contexts/StyleErrorsContext.js [hunk 3]
```diff
@@ -12,26 +12,86 @@
...
+			if (state[itemId]?.[fieldName]) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/contexts/StyleErrorsContext.js:34]: This logic is a bit hard to follow. Do you think we can simplify it somehow?

If have tried to think about a simpler solution but can't imaging anything. Maybe that one is the best.

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/ColorPicker/ColorPicker.js [hunk 4]
```diff
@@ -302,9 +324,10 @@ export function ColorPicker({
 												ref={inputRef}
 												role="combobox"
 												value={
-													color.startsWith('#')
+													error.value ||
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/ColorPicker/ColorPicker.js:327]:
```
I think
        we are starting
                to have
                    too
                    much
                            nes
                            ting
                            in
                            this
                            compone
                            nt
```

**Review by p2kmgcl — COMMENTED** [review]
I added some comments to things that we might want to update, but the PR looks really nice. Much better to keep track of which field has X error :)

**Comment by Tim-Cao** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
ci:forward
